data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/prepper/peerforwarder/discovery/AwsCloudMapPeerListProvider.java

```diff
@@ -0,0 +1,205 @@
+package com.amazon.dataprepper.plugins.prepper.peerforwarder.discovery;
+
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.plugins.prepper.peerforwarder.PeerForwarderConfig;
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.client.endpoint.DynamicEndpointGroup;
+import com.linecorp.armeria.client.retry.Backoff;
+import com.linecorp.armeria.common.CommonPools;
+import io.netty.channel.EventLoop;
+import io.netty.util.concurrent.ScheduledFuture;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.servicediscovery.ServiceDiscoveryAsyncClient;
+import software.amazon.awssdk.services.servicediscovery.model.DiscoverInstancesRequest;
+import software.amazon.awssdk.services.servicediscovery.model.DiscoverInstancesResponse;
+import software.amazon.awssdk.services.servicediscovery.model.HttpInstanceSummary;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of {@link PeerListProvider} which uses AWS CloudMap's
+ * service discovery capability to discover peers.
+ */
+class AwsCloudMapPeerListProvider implements PeerListProvider, AutoCloseable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AwsCloudMapPeerListProvider.class);
+    private static final int ONE_SECOND = 1000;
+    private static final int TWENTY_SECONDS = 32000;
```

Comment:
nit: These can be rename to: INITIAL_DELAY_MILLIS & MAX_DELAY_MILLIS

Reply:
I took this approach because I wanted it to be clear when reading the code.


```
final Backoff standardBackoff = Backoff.exponential(ONE_SECOND, TWENTY_SECONDS).withJitter(TWENTY_PERCENT);
```

vs.

```
final Backoff standardBackoff = Backoff.exponential(INITIAL_DELAY_MILLIS, MAX_DELAY_MILLIS).withJitter(TWENTY_PERCENT);
```

I find the former clearer because you look at it and immediately know what the range is.

Reply:
Sounds good